src/common/time/instant.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {TimeUnit} from './timeunit.js';
+
+/**
+ * An Instant represents a specific instance of universal time with a
+ * defined precision.
+ *
+ * TODO implement comparable?
+ * TODO implement other utility methods
+ */
+export class Instant {
+  private readonly secs: number;
+  private readonly nanos: number;
+  private readonly resolutionValue: TimeUnit;
+  
+  private constructor(secs: number, nanos = 0, resolution: TimeUnit) {
```

Comment:
I like the pattern for arguments that's required args first, default argument last.

I think we should either: 
1. make nanos the last argument
2. give resolution a default argument

Reply:
default arg seems fine.  I'm less concerned with the constructor API since it's a bit private implementation detail.